### DISCUSSION TITLE
Welcome to rna-tools Discussions!

### DISCUSSION THREAD

**Discussion by mmagnus** [mmagnus/rna-tools#111] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by wkdawson** [comment]
Good to hear from you Marcin. It's a thankless and tedious task developing software, so thank you for doing what you can. I've been working in a rather orthogonal direction of late, so I haven't been able to contribute much of anything, due to too many distractions. Anyway, I hope I can at least encourage you to keep working at it.

**Reply by mmagnus** [reply]
Yeah, it's very hard, and not very appreciated in science (this is how I feel ;-)).

Anyway, for me is always self-motivating to do the code right because other people might see it. This really helps me to get my focus and put more effort into a code. And often it's me from the future that benefits the most when I'm trying to run something after months... if it's written well, then it saves me a lot of time. If it's poorly written, then I pretty much have to re-write it from scratch. Not very efficient way.

And if you have lots of code well written then actually it kind of helps you to build new software on top of that. So this is also super fun.

Thank you for your words! :-) and yeah, join the team of developers or testers any time.

**Comment by jessicablaisekelly** [comment]
Hi Marcin,
I'm very sorry to bother you with such a rookie question, I am not very familiar with Python but I am using Pymol and was trying to add rna tools, but I am hitting a lot of issues. I've carefully followed the documentation both using git clone and by trying to add via the plugin manager in pymol, but I keep getting an error that the plugin has been installed but is not initialised, saying it cannot find a module called icecream.

I know this is probably a very silly question, but can you please help as I really need to be able to model RNA for my phd project and have spent hours on this so far.

Thank you in advance.

**Reply by mmagnus** [reply]
Hey, what's the error msg?

**Reply by jessicablaisekelly** [reply]
Hi Marcin ,

Thank you for getting back to me so quickly! I have copied in the log from the command line, I hope this helps to explain my issue better!
Traceback (most recent call last):
  File "/Applications/PyMOL.app/Contents/lib/python3.10/site-packages/pymol/parsing.py", line 470, in run
    run_(path, ns_pymol, ns_pymol)
  File "/Applications/PyMOL.app/Contents/lib/python3.10/site-packages/pymol/parsing.py", line 519, in run_file
    execfile(file,global_ns,local_ns)
  File "/Applications/PyMOL.app/Contents/lib/python3.10/site-packages/pymol/parsing.py", line 514, in execfile
    exec(co, global_ns, local_ns)
  File "/Users/jessicakelly/src/rna-tools/rna_tools/tools/PyMOL4RNA/PyMOL4RNA.py", line 2, in <module>
    from icecream import ic
ModuleNotFoundError: No module named 'icecream'

**Reply by mmagnus** [reply]
can you type in the terminal to install missing python module, icecream

pip install icecream

and tell me if this worked

**Reply by jessicablaisekelly** [reply]
Unfortunately this didn't work, when I said that it said that it's already installed. However, when I go to pymol to install the plugin it allows me to install it but then says it's unable to initialise the plugin: 
Unable to initialize plugin 'PyMOL4RNA' (pmg_tk.startup.PyMOL4RNA).

So I entered in the following: 
PyMOL>PyMOL4RNA_PATH='/Users/jessicakelly/src/rna-tools/rna_tools/tools/PyMOL4RNA/PyMOL4RNA.py'; cmd.run(PyMOL4RNA_PATH)

And then I get the following error: 
Traceback (most recent call last):
  File "/Applications/PyMOL.app/Contents/lib/python3.10/site-packages/pymol/parsing.py", line 470, in run
    run_(path, ns_pymol, ns_pymol)
  File "/Applications/PyMOL.app/Contents/lib/python3.10/site-packages/pymol/parsing.py", line 519, in run_file
    execfile(file,global_ns,local_ns)
  File "/Applications/PyMOL.app/Contents/lib/python3.10/site-packages/pymol/parsing.py", line 514, in execfile
    exec(co, global_ns, local_ns)
  File "/Users/jessicakelly/src/rna-tools/rna_tools/tools/PyMOL4RNA/PyMOL4RNA.py", line 2, in <module>
    from icecream import ic
ModuleNotFoundError: No module named 'icecream'

I'm not sure if I'm installing the plugin the correct way? As it seems to be fine up until that last step.

**Reply by mmagnus** [reply]
Can you try git pull again?
